python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+                                 photParams=photParams)
+        return magerr[0]
+
+    def catsimBandFluxError(self, time, bandpassobject, m5, photParams=None,
+                            magnitude=None):
+        """
+        return the flux uncertainty in the bandpass in units 'maggies'
+        (the flux the AB magnitude reference spectrum would have in the
+        same band.)
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassobject: mandatory, `lsst.sims.photUtils.BandPass` object
+            A particular bandpass which is an instantiation of one of
+            (u, g, r, i, z, y)
+        m5 :
+        photParams :
+        magnitude :
+        Returns
+        -------
+        float
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        # SEDfromSNcosmo = self.SNObjectSED(time=time,
+        #                                 bandpass=bandpassobject)
+
+        if magnitude is None:
+            mag = self.catsimBandMags(time=time, bandpassobject=bandpassobject)
+        else:
+            mag = magnitude
+        flux = self.catsimBandFluxes(time=time, bandpassobject=bandpassobject)
+
+        # mag = np.asarray([[mag]])
+        bandpass = bandpassobject
+
+        if photParams is None:
+            photParams = PhotometricParameters()
+
+        # m5 = np.asarray([[m5]])
+        SNR, gamma = calcSNR_m5(magnitude=mag, bandpass=bandpass, m5=m5,
+                                photParams=photParams)
+        return flux / SNR
+
+    def catsimManyBandFluxes(self, time, bandpassDict,
+                             observedBandPassInd=None):
+        """
+        return the flux in the bandpass in units of the flux
+        the AB magnitude reference spectrum would have in the
+        same band.
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassDict: mandatory, `lsst.sims.photUtils.BandpassDict` instance
+        observedBandPassInd : optional, list of integers, defaults to None
+            integer correspdonding to index of the bandpasses used in the
+            observation in the ordered dict bandpassDict
+        Returns
+        -------
+        `~numpy.ndarray` of length =len(observedBandPassInd)
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        SEDfromSNcosmo = self.SNObjectSED(time=time,
+                                          bandpass=bandpassDict['u'])
+        wavelen_step = np.diff(SEDfromSNcosmo.wavelen)[0]
+        SEDfromSNcosmo.flambdaTofnu()
+        f = SEDfromSNcosmo.manyFluxCalc(bandpassDict.phiArray,
```

Comment:
bandpassDict has its own equivalent of manyFluxCalc (BandpassDict.fluxListForSed).  Why aren't you using that?  Using the BandpassDict method will probably make this code easier to maintain after our sims_photUtils overhaul, as the BandpassDict API is much more likely to remain stable than is the Sed.manyFluxCalc API.